### PR TITLE
Iterative training hotfix

### PR DESCRIPTION
**Pull request by yrkim98**
- This changes regular prediction runs to override the correct ckpt_path key in the cyto-dl config
- Also explicitly turns off iterative training- so we dont have to rely on cyto-dl configs to have iterative training turned off by default
- Update tests

### CODE REVIEW

File: src/allencell_ml_segmenter/utils/cyto_overrides_manager.py [hunk 1]
```diff
@@ -62,6 +62,9 @@ def get_training_overrides(
                 )
             # Filters/Model Size (required if starting new model)
             overrides_dict["model._aux.filters"] = model_size.value
+            # needed to specify no weights used
+            overrides_dict["checkpoint.weights_only"] = False
```
Comment (hughes036) [inline src/allencell_ml_segmenter/utils/cyto_overrides_manager.py:66]: how does `checkpoint.weights_only` get overidden?

Comment (hughes036) [reply src/allencell_ml_segmenter/utils/cyto_overrides_manager.py:66]: was that already implemented ?

Comment (hughes036) [reply src/allencell_ml_segmenter/utils/cyto_overrides_manager.py:66]: Does the config itself have a default?  It seems like if we want to default to False , that would be the ideal place to do it.

Comment (saeliddp) [reply src/allencell_ml_segmenter/utils/cyto_overrides_manager.py:66]: Looks like that happens here: [in the overrides manager class](https://github.com/AllenCell/allencell-segmenter-ml/blob/5676077d5f81e3c519e18bcea9bdf5c98dfd8938/src/allencell_ml_segmenter/utils/cyto_overrides_manager.py#L42)

**Review by saeliddp — APPROVED** [review]

File: src/allencell_ml_segmenter/utils/cyto_overrides_manager.py (hunk 1 above)
Comment (hughes036) [reply src/allencell_ml_segmenter/utils/cyto_overrides_manager.py:66]: Is there test cov for `weights_only` ?

Comment (hughes036) [reply src/allencell_ml_segmenter/utils/cyto_overrides_manager.py:66]: in `test_get_training_overrides_iterative_training` i see as asserts True on `weights_only` but i dont see anything setting `is_using_existing_model` in the ARRANGE section.

Comment (yrkim98) [reply src/allencell_ml_segmenter/utils/cyto_overrides_manager.py:66]: `checkpoint.weights_only` should be True for iterative training runs and False for regular training runs.

The config's defaults are `checkpoint.weights_only = False` and `checkopint.strict = True` so we shouldnt need to explicitly set here, just wanted to ensure it does get set for non iterative training runs. I will remove this.

Comment (yrkim98) [reply src/allencell_ml_segmenter/utils/cyto_overrides_manager.py:66]: > in `test_get_training_overrides_iterative_training` i see as asserts True on `weights_only` but i dont see anything setting `is_using_existing_model` in the ARRANGE section.

In this test we are setting `model.set_is_using_existing_model(True)` when arranging on line 117.

Comment (hughes036) [reply src/allencell_ml_segmenter/utils/cyto_overrides_manager.py]: Thanks for pointing out 117, i missed it .  lgtm.

**Review by hughes036 — APPROVED** [review]

File: src/allencell_ml_segmenter/utils/cyto_overrides_manager.py [hunk 2]
```diff
@@ -62,6 +62,9 @@ def get_training_overrides(
...
+            overrides_dict["checkpoint.strict"] = True
```
[the remark this answers is not in the document]
Comment (benjijamorris) [reply src/allencell_ml_segmenter/utils/cyto_overrides_manager.py]: strict means that every weight's name and shape in the checkpoint file has to match the model names/shapes that it's being loaded into

**Pull request merged**